### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
Updated PR to remove one commit that you wouldn't like.  (it changed `size_t i` to `uint32_t`.  I'll fix it and submit it as part of the next round of warning fixes)

File: events/events_network.c [hunk 1]
```diff
@@ -255,7 +255,7 @@ events_network_select(struct timeval * tv)
 	events_network_selectstats_select();
 
 	/* Select. */
-	while (select(socketlist_getsize(S), &readfds, &writefds,
+	while (select((int) socketlist_getsize(S), &readfds, &writefds,
```
Comment (cperciva) [inline events/events_network.c]: Unnecessary whitespace in cast.

File: events/events_timer.c [hunk 2]
```diff
@@ -99,8 +99,8 @@ events_timer_register_double(int (*func)(void *), void * cookie,
 	struct timeval tv;
 
 	/* Convert timeo to a struct timeval. */
-	tv.tv_sec = timeo;
-	tv.tv_usec = (timeo - tv.tv_sec) * 1000000.0;
+	tv.tv_sec = (time_t) timeo;
+	tv.tv_usec = (suseconds_t) ((timeo - tv.tv_sec) * 1000000.0);
```
Comment (cperciva) [inline events/events_timer.c]: Unnecessary whitespace in cast (twice).

File: util/b64encode.c [hunk 3]
```diff
@@ -97,7 +97,7 @@ b64decode(const char * in, size_t inlen, uint8_t * out, size_t * outlen)
 
 		/* Output 3 bytes. */
 		for (i = 0; i < 3; i++) {
-			out[i] = t >> 16;
+			out[i] = (uint8_t) (t >> 16);
```
Comment (cperciva) [inline util/b64encode.c]: Unnecessary whitespace in cast.  And I'd prefer `(t >> 16) & 0xff` if that silences the warning.

File: util/sysendian.h [hunk 4]
```diff
@@ -22,7 +22,7 @@ be16dec(const void * pp)
 {
 	const uint8_t * p = (uint8_t const *)pp;
 
-	return ((uint16_t)(p[1]) + ((uint16_t)(p[0]) << 8));
+	return (uint16_t) ((uint16_t)(p[1]) + ((uint16_t)(p[0]) << 8));
```
Comment (cperciva) [inline util/sysendian.h]: Unnecessary whitespace in cast.

File: util/sysendian.h [hunk 5]
```diff
@@ -85,7 +85,7 @@ le16dec(const void * pp)
 {
 	const uint8_t * p = (uint8_t const *)pp;
 
-	return ((uint16_t)(p[0]) + ((uint16_t)(p[1]) << 8));
+	return (uint16_t) ((uint16_t)(p[0]) + ((uint16_t)(p[1]) << 8));
```
Comment (cperciva) [inline util/sysendian.h]: Unnecessary whitespace in cast.

**Comment by gperciva** [comment]
Updated to fix whitespace issues, and convert to int with `& 0xff`.
